pkg/connection/store/external/store.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package external implements a gRPC client for external secret store plugins.
+package external
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"path/filepath"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	ess "github.com/crossplane/crossplane-runtime/apis/proto/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/connection/store"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+// Error strings.
+const (
+	errGet    = "cannot get secret"
+	errApply  = "cannot apply secret"
+	errDelete = "cannot delete secret"
+)
+
+// SecretStore is an External Secret Store.
+type SecretStore struct {
+	client     ess.ExternalSecretStoreServiceClient
+	kubeClient client.Client
+	config     *v1.Config
+}
+
+// NewSecretStore returns a new External SecretStore.
+func NewSecretStore(ctx context.Context, kube client.Client, tlsConfig *tls.Config, cfg v1.SecretStoreConfig) (*SecretStore, error) {
+	if cfg.Plugin.Endpoint == nil {
+		return nil, errors.New("endpoint is not provided")
```

Comment:
Should these error strings be constant? It seems like some are (e.g. `errGet`). I like having error strings be constants because you can then change them without having to update them in two places (i.e. the code and the tests).